src/coprocessor/dag/expr/builtin_math.rs

```diff
@@ -154,6 +154,22 @@ impl ScalarFunc {
         }
         Ok(Some(pow))
     }
+
+    #[inline]
+    pub fn truncate_int(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let x = try_opt!(self.children[0].eval_int(ctx, row));
+        let d = try_opt!(self.children[1].eval_int(ctx, row));
+        Ok(Some((truncate(x as f64, d)) as i64))
```

Comment:
How about using a lookup table?

Reply:
hi, I did not get what do you mean 'lookup table' here,  could you please give me more info?